### PR TITLE
[Terra-Date-Time-Picker] - Added visual focus dashed border and SR fix

### PR DESCRIPTION
**Pull request by PK106552**
### Summary

**What was changed:**

1. Added visual focus dashed border.
3. Added errorId prop in `terra-date-time-picker` example for invalid error message announcement.

**Why it was changed:**

1. Previously visual focus has blue border which is not valid.
3. Error message was not reading by screen reader.

### Testing

This change was tested using:

- [ ] WDIO
- [ ] Jest
- [x] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [x] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-XXXX 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

**Review by saket2403 — APPROVED** [review]

File: packages/terra-date-picker/src/DateInput.jsx [hunk 1]
```diff
@@ -861,7 +869,7 @@ const DatePickerInput = (props) => {
           onBlur={onBlur}
           onFocus={onButtonFocus}
           refCallback={buttonRefCallback}
-          aria-label={`${calendarDate} ${intl.formatMessage({ id: 'Terra.datePicker.openCalendar' })}`}
```
Comment (sugan2416) [inline packages/terra-date-picker/src/DateInput.jsx:864]: If this is removed, then SR will not announce the date selected from calendar popup upon closure. This needs to discussed with a11y team. This date is to provide date context due to instant closing of calendar.

File: packages/terra-date-time-picker/src/DateTimePicker.jsx [hunk 2]
```diff
@@ -135,6 +135,11 @@ const propTypes = {
    * If the `variant` prop if set to `12-hour` for one of these supported locales, the variant will be ignored and defaults to `24-hour`.
    */
   timeVariant: PropTypes.oneOf([DateTimeUtils.FORMAT_12_HOUR, DateTimeUtils.FORMAT_24_HOUR]),
+  /**
+   * If invalid error text is used, provide a string containing the IDs for error html element.
+   * ID must be htmlFor prop value with error text.
+   */
+  ariaDescribedBy: PropTypes.string,
```
[the remark this answers is not in the document]
Comment (PK106552) [reply packages/terra-date-time-picker/src/DateTimePicker.jsx]: Updated - https://github.com/cerner/terra-framework/pull/2035/commits/381aac0a254933c1df687af6c65661d174f05601

**Review by sugan2416 — APPROVED** [review]

**Comment by PK106552** [comment]
Hi rahul, Reverted selected announcement fixes and created new JIRA for that - https://jira2.cerner.com/browse/UXPLATFORM-10245

**Comment by rahulkumar8cs** [comment]
+1 for the a11y changes.

**Pull request merged**